web/src/stream_settings_ui.ts

```diff
@@ -1178,4 +1179,8 @@ export function initialize(): void {
         $(".right").removeClass("show");
         $("#channels_overlay_container .two-pane-settings-header").removeClass("slide-left");
     });
+
+    $("#channels_overlay_container").on("click", "#preview-stream-button", () => {
+        window.location.href = preview_url;
```

Comment:
Instead of storing url in a local variable, we can get stream ID from `.stream_setings_header` element and then pass it to `hash_util.by_stream_url`.